### DISCUSSION TITLE
What if  isLoading & isSuccess same time ?

### DISCUSSION THREAD

**Discussion by iamyoki** [TanStack/query#2320] [Q&A]
## Pain Point:
Sometimes when we click tab button and fetch new tab data, we don't want the old tab suddenly becomes to blank page while fetching, we want to wait till the end of fetching then fresh the old tab page to new one.

So, how can I do that?

**Comment by TkDodo** [comment]
I don't understand the question, but judging from the title: you cannot be in loading and success state at the same time. The booleans are derived from the `status` field and the internal state machine, so this cannot happen.

it sounds like you want `keepPreviousData` maybe?
